### PR TITLE
Address Coveralls Reported Coverage Level

### PR DESCRIPTION
**Pull request by anneal**
## Description
This PR addresses Issue https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/issues/190

My guess is that the way Goveralls [merges profiles](https://github.com/mattn/goveralls/blob/e68c968003c2db2ad7cdf07b3c012c5ab12248a7/gocover.go#L32-L34) may have been affected by one of the Go releases. 

This PR uses the functionality of [Go 1.10](https://golang.org/doc/go1.10#test) to create the coverprofile. 
>The go test -coverpkg flag now interprets its argument as a comma-separated list of patterns to match against the dependencies of each test, not as a list of packages to load anew.

There are some existing issues open with how `coverpkg` works which may change the behavior of `go test` coverage reports in the future:
- https://github.com/golang/go/issues/23910
- https://github.com/golang/go/issues/23883

## Screenshots
Coverage prior to this PR:
screen shot 2018-08-04 at 8 29 29 am
https://coveralls.io/builds/18322614

Coverage with this PR:
screen shot 2018-08-04 at 8 31 18 am
https://coveralls.io/builds/18325241

### CODE REVIEW

**Comment by anneal** [comment]
I signed it!

**Comment by G-Harmon** [comment]
Hey April,
Thank you for helping out!  The automated test failed (https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/GoogleCloudPlatform_k8s-multicluster-ingress/200/pull-kubernetes-multicluster-ingress-test/256/), but it also failedl like this on the last PR I saw.  What concerns me is that there might be something more fundamentally wrong than what's in #190.

0803 19:24:01.716] Bad response status from coveralls: 422
W0803 19:24:01.716] {"message":"Couldn't find a repository matching this job.","error":true}

@nikhiljindal, thoughts?

**Comment by anneal** [comment]
I noticed the build failure on the other PR too! Maybe the token in `/etc/coveralls-token/coveralls.txt` isn't populating the make target properly?

**Comment by G-Harmon** [comment]
I tested goveralls locally with:
goveralls -v -repotoken <redacted>
and it passed: http://coveralls.io/jobs/39004044

@BenTheElder, can you check on our Goveralls token configuration? Specifically, is our /etc/coveralls-token/coveralls.txt secret still look valid? 
I think it's configured here: https://github.com/kubernetes/test-infra/blob/af23d941eb0b6d615d3a1360376827db5b3f678a/prow/config.yaml#L520

**Comment by BenTheElder** [comment]
```
$ kubectl describe secret -n=test-pods k8s-multicluster-ingress-coveralls-token
Name:         k8s-multicluster-ingress-coveralls-token
Namespace:    test-pods
Labels:       <none>
Annotations:  <none>

Type:  Opaque

Data
====
coveralls.txt:  <elided by bentheelder>
```

your secret hasn't changed and that mount looks fine. I don't know anything about the token itself.

**Comment by anneal** [comment]
I'm just saying my thoughts out loud here as I look at how this is all connected...

We're using this image for the tests:
`gcr.io/k8s-testimages/kubekins-e2e:v20180730-8b7ab3104-master`
Repo here:
https://github.com/kubernetes/test-infra/tree/master/images/kubekins-e2e

The most recent change bumped the docker bootstrap version, but it could be any change between April 18th [(last passing build)](https://coveralls.io/builds/16546664) and July 26th. 

It looks like a next debugging step could be pulling down the test-infra repo and running the test using the kubekins-e2e Dockerfile. I am also willing to back away slowly if anyone else picks this up and solves it in the meantime.

**Comment by G-Harmon** [comment]
I'm testing a potential fix in https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/pull/201
https://docs.coveralls.io/supported-ci-services makes it look like we're now missing CI_BUILD_URL as a environment variable.

**Comment by G-Harmon** [comment]
Good find on the "-service" flag! I also got an email showing it worked. It pointed to https://coveralls.io/builds/18322381

Can you please submit a separate PR that just does the "-service" fix? That seems worthy of going in on its own.

**Comment by G-Harmon** [comment]
You can send a PR, or you can LGTM #201, where I did just the "-service" flag.

**Comment by anneal** [comment]
Is squashing commits on a PR acceptable in this repo? I'd like to combine 
https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/pull/200/commits/f98a0f46243314b5bd426c2a8247bed0e2646527
https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/pull/200/commits/ed3bd56b4189a77224dad560fc8247e13f40e661
https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/pull/200/commits/6d98247db82babeaa344c8d1f8c2abbaf5af0c8b
https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/pull/200/commits/3b29b2a1cac5eec11eca25aea594a3945b3e6b78

Code coverage is back up to 56%. I recall something about coverage profiles across packages, but it might take another few days for me to have time to dig in farther. It appears that the `test/e2e/case/util.go` is covering `app/kubemci/pkg/goutils`, but getting that to show in a profile report doesn't immediately occur to me.

File: Makefile [hunk 1]
```diff
@@ -42,9 +43,18 @@ cover:
 
 coveralls:
 	@echo "+ $@"
+# Generate coverage report
+	@go test \
+		-coverprofile=${COVERPROFILE} \
+		-coverpkg=$(shell go list ${PKG}/... | xargs | sed -e 's/ /,/g') \
```
Comment (G-Harmon) [inline Makefile]: Thanks for digging into this and finding a solution!

But, how does this command differ from the one in "cover" above? Seems like we should be able to use basically an identical command...?

Comment (anneal) [reply Makefile]: I'm not finding anything using the `cover` target so it makes sense to update that one.

**Comment by G-Harmon** [comment]
/lgtm
looks good, thanks! Glad to see the reuse between "cover" and "coveralls".

**Comment by G-Harmon** [comment]
thanks again for figuring this out.

**Pull request merged**